updated the whole Contribute to Kotlin section, added information on EAP (#1603)
JetBrains/kotlin-web-site: pages/contribute.md

If you want to propose changes to a specific page, you can click **Edit Page** right on the page.
  
The documentation site is built using Markdown and Jekyll. 

## Create tutorials or videos

If you've created tutorials or videos for Kotlin, please share them with us. We would be happy to add them to [Awesome Kotlin](https://kotlin.link/) maintained by the community.

## Translate documentation to other languages

You are welcome to translate the Kotlin documentation into your own language and publish your translation on your web site.
However, we won't be able to host your translation in the main repository and publish it on [kotlinlang.org](https://kotlinlang.org/docs/reference/).

This site is the official documentation for the language, and we ensure that all the information here
is correct and up to date. Unfortunately, we won't able to review documentation in your own language. 

## Hold events and presentations

If you've given or just plan to give presentations or hold events on Kotlin, let us know. We'll feature them on [the event list](http://kotlinlang.org/docs/events.html)

> ##Participate in Early Access Preview

Reviewer: ```suggestion
## Participate in Early Access Preview

```

# Contributing to Kotlin

> Kotlin is an Apache 2 OSS Project. The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). While Kotlin is mostly developed by JetBrains, there are hundreds of external contributors to the Kotlin project, and we always look for more people&mdash;become one of them. 

Author: Agree. Added the link to Community.

> Kotlin is an Apache 2 OSS Project. The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). While Kotlin is mostly developed by JetBrains, there are hundreds of external contributors to the Kotlin project, and we always look for more people&mdash;become one of them. 

Author: Discussed with Alina. She doesn't recommend adding CTA here -- there will be a closed loop for users. 
Rephrased.

## Contribute to the code

If you want to contribute to the Kotlin code, go to [GitHub](https://github.com/jetbrains/kotlin), check out the latest Kotlin version,
 and follow the [instructions on how to contribute to the code](https://github.com/JetBrains/kotlin/blob/master/docs/contributing.md).

> You can complete [open tasks](https://youtrack.jetbrains.com/issues/KT?q=tag:%20%7BUp%20For%20Grabs%7D%20and%20State:%20Open) or suggest an improvement using the [Kotlin Evolution and Enhancement Process (KEEP)](https://github.com/Kotlin/KEEP).

Author: Updated to be more specific. Removed information about KEEP. We promote only solving YT tasks. Advanced users will find ways to contribute themselves.

## Contribute to existing libraries and create your own

> Kotlin ships with the standard library providing limited capabilities and a number of additional (kotlinx) libraries that extend its functionality. 

Author: Agree. Will use "core" instead.

> Each kotlinx library is stored in a separate repo, has its own versioning and release cycle. 

Author: OK

> If you want to contribute to the standard library, follow the process for contributing to the Kotlin code.

Author: Agree. Added.

If you want to contribute to a kotlinx library, follow the contribution process described for the library&mdash;[kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines),

>  [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/CONTRIBUTING.md), and [ktor](https://github.com/ktorio/ktor/blob/master/CONTRIBUTING.md).

Author: Discussed with Stas. We don't want to encourage people to contribute to unstable libraries. That's why we have decided to list only the stable ones.

If you have a library that could be useful for others, let us know.

## Contribute to the documentation

> If you have helpful information for [Kotlin documentation](https://kotlinlang.org/docs/reference/) or have found a documentation issue, feel free to check out [the documentation source code on GitHub](https://github.com/jetbrains/kotlin-web-site),

Author: You are right. Updated

> commit changes, and send us a Pull Request. Please stay in touch because we may have questions and comments on your changes. Otherwise, we won't be able to incorporate them. 

Author: It is relevant for code as well. I have added. Thx for pointing out!

If you want to propose changes to a specific page, you can click **Edit Page** right on the page.
  

> The documentation site is built using Markdown and Jekyll. 

Author: It is a good point. Thx! I will ask Pavel if we have any contribution guildlines. I haven't found myself.
Removed for now. Will add a link to guidelines once we have them.

## Create tutorials or videos

> If you've created tutorials or videos for Kotlin, please share them with us. We would be happy to add them to [Awesome Kotlin](https://kotlin.link/) maintained by the community.

Reviewer 2: Same question: how do I share anything with you?

## Translate documentation to other languages

You are welcome to translate the Kotlin documentation into your own language and publish your translation on your web site.
However, we won't be able to host your translation in the main repository and publish it on [kotlinlang.org](https://kotlinlang.org/docs/reference/).

This site is the official documentation for the language, and we ensure that all the information here

> is correct and up to date. Unfortunately, we won't able to review documentation in your own language. 

Reviewer 2: > won't able

typo: won't be able

Reviewer 2: Also, I'm not sure "your own language" sounds polite enough. Does it?

Author: Fixed.
Changed "your own language" to "other languages"

## Hold events and presentations

> If you've given or just plan to give presentations or hold events on Kotlin, let us know. We'll feature them on [the event list](http://kotlinlang.org/docs/events.html)

Reviewer 2: Same question :)

Author: Waiting for Alina to provide the link.

> ## Participate in Early Access Preview

Reviewer 2: Looks like this won't be visible to very many readers. Should we move it earlier?

Author: Moved to the top

# Contributing to Kotlin

> Kotlin is an Apache 2 OSS Project. The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). 

Reviewer 2: ```suggestion
Kotlin is an open-source project under the [Apache 2.0 License](https://github.com/JetBrains/kotlin/blob/master/license/LICENSE.txt). The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). 
```

> Kotlin is an Apache 2 OSS Project. The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). 

Author: fixed. thx

> While Kotlin is mostly developed by JetBrains, there are hundreds of external contributors to the Kotlin project, and we always look for more people&mdash;[become one of them](https://kotlinlang.org/community). 

Reviewer 3: and we are always on the lookout for more people to help us
and we are always welcome to having new people 
and we always like to have new people

but we always welcome more

or cut this bit entirely to:
While Kotlin is mostly developed by JetBrains, there are hundreds of external contributors to the Kotlin project [become one of them](https://kotlinlang.org/community).

Author: Decided to shorten. Thx for options!

## Participate in Early Access Preview

> You can help us improve Kotlin if you participate in Kotlin Early Access Preview (EAP) and provide your valuable feedback.

Reviewer 3: You can help us improve Kotlin by participating in Kotlin Early Access Preview (EAP) and providing your valuable feedback.

Author: Done

> For every release, Kotlin ships a few preview builds where you can try out the latest features before they go to production. You can report found bugs in [YouTrack](https://kotl.in/issue) 

Reviewer 3: You can report any bugs you find to our issue tracker [YouTrack](https://kotl.in/issue)

Author: Done

> and we will try to fix them in the next EAP build. This way, you can get bug fixes earlier than the standard Kotlin release cycle.

Reviewer 2: ```suggestion
and we will try to fix them before a final release. This way, you can get bug fixes earlier than the standard Kotlin release cycle.
```

Reviewer 2: It can be not in an EAP build but in the actual release build

## Contribute to the compiler, standard library, and tooling

If you want to contribute to the Kotlin compiler, standard library, or tooling, go to [GitHub](https://github.com/jetbrains/kotlin), check out the latest Kotlin version,
 and follow the [instructions on how to contribute to the code](https://github.com/JetBrains/kotlin/blob/master/docs/contributing.md).
 
You can help us by completing [open tasks](https://youtrack.jetbrains.com/issues/KT?q=tag:%20%7BUp%20For%20Grabs%7D%20and%20State:%20Open). 

> Please stay in touch with us because we may have questions and comments on your changes. Otherwise, we won't be able to incorporate them. 

Reviewer 3: Please keep an open line of communication with us because we may have questions and comments on your changes. Otherwise, we won't be able to incorporate your contributions.

Author: done

## Contribute to other libraries and create your own

> Kotlin ships with the standard library providing core capabilities and a number of additional (kotlinx) libraries that extend its functionality. 

Reviewer 2: ```suggestion
Besides the standard library that provides core capabilities, Kotlin has a number of additional (kotlinx) libraries that extend its functionality. 
```

Reviewer 2: The standard library was mentioned in the heading above, and it's too late to explain that it exists :) Also, we don't ship `kotlinx` libraries with the Kotlin distribution, unlike the stdlib

> Besides the standard library that provides core capabilities, Kotlin has a number of additional (kotlinx) libraries that extend its functionality. 

Author: I would not use this special naming because not all library names include kotlinx, for example, ktor.

Each kotlinx library is developed in a separate repository, has its own versioning and release cycle.

If you want to contribute to a kotlinx library, follow the contribution process described for each library, such as [kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines),
 [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/CONTRIBUTING.md), and [ktor](https://github.com/ktorio/ktor/blob/master/CONTRIBUTING.md).

> If you have a library that could be useful for other Kotlin developers, let us know via <feedback@kotlinlang.org>.

Reviewer 2: ```suggestion
If you have a library that could be useful to other Kotlin developers, let us know via <feedback@kotlinlang.org>.
```

> If you have a library that could be useful for others, let us know via <feedback@kotlinlang.org>.

Reviewer 3: If you have a library that could be useful for other Kotlin developers

Author: done

## Contribute to the documentation

> If you have helpful information for [Kotlin documentation](https://kotlinlang.org/docs/reference/) or have found a documentation issue, 

Reviewer 3: If you have information that would be helpful to add to the Kotlin documentation or have found an issue in the documentation,

-- actually this is totally your call, i would only have it this way as it feels more complete, but feel free to not use this.

Author: We will keep info only about issues in docs for now. Will add contribution guidelines later. Right now they don't exist.

feel free to check out [the documentation source code on GitHub](https://github.com/JetBrains/kotlin-web-site/tree/master/pages/docs)

> and send us a Pull Request. Please stay in touch with us because we may have questions and comments on your changes. Otherwise, we won't be able to incorporate them. 

Reviewer 3: Please keep an open line of communication with us because we may have questions and comments on your changes. And otherwise, we won't be able to incorporate your changes.

Author: done

The documentation site is built using Markdown and Jekyll. 

## Create tutorials or videos

> If you've created tutorials or videos for Kotlin, please share them with us via <feedback@kotlinlang.org>. We would be happy to add them to [Awesome Kotlin](https://kotlin.link/) maintained by the community.

Reviewer 3: Is it called Awesome Kotlin?

Kotlin is Awesome  ??

If you've created tutorials or videos for Kotlin, please share them with us via <feedback@kotlinlang.org>. We'd be happy to add them to the list of resources at [Kotlin is Awesome](https://kotlin.link/), which is maintained by the community

Author: done

## Translate documentation to other languages

> You are welcome to translate the Kotlin documentation into your own language and publish your translation on your web site.

Reviewer 3: You are welcome to translate the Kotlin documentation into your own language and publish the translation on your website.

Author: done

# Contributing to Kotlin

Kotlin is an Apache 2 OSS Project. The source code, tooling, documentation and even this web site is maintained on [GitHub](https://github.com/jetbrains/kotlin). 

> While Kotlin is mostly developed by JetBrains, there are hundreds of external contributors to the Kotlin project&mdash;[become one of them](https://kotlinlang.org/community). 

Reviewer 4: We shouldn't link the reader from here to the community section. There is no additional information for the contributors at the page and, we'll put the user on a closed loop.  
The paragraph also is introductory and recaps many different possible types of contribution, thus making it quite hard to point a reader to one place where they will become a contributor. In fact we will point the reader to many different places within further paragraphs of the guideline. That's why I think we don't need a CTA here at all.
My suggestion is we rephrase it somewhat like: "there are hundreds of external contributors to the Kotlin and you can find out how to become one of them in the following guidelines"

As an idea, at some point, we can provide a link to the list of contributors on GitHub https://github.com/JetBrains/kotlin/graphs/contributors, however, it will not represent those who contribute, for example, learning materials.

Author: Got your point! Removed CTA.

## Participate in Early Access Preview

> You can help us improve Kotlin by participating in Kotlin Early Access Preview (EAP) and providing us with your valuable feedback.

Reviewer 4: Am I correct, you will add here instruction on how to get access to EAP later? @katia-energizer

Author: That's right. I will add a link once we have articles published on the website.

and we will try to fix them in the next EAP build. This way, you can get bug fixes earlier than the standard Kotlin release cycle.

## Contribute to the compiler, standard library, and tooling

If you want to contribute to the Kotlin compiler, standard library, or tooling, go to [GitHub](https://github.com/jetbrains/kotlin), check out the latest Kotlin version,
 and follow [the instructions on how to contribute](https://github.com/JetBrains/kotlin/blob/master/docs/contributing.md).
 
You can help us by completing [open tasks](https://youtrack.jetbrains.com/issues/KT?q=tag:%20%7BUp%20For%20Grabs%7D%20and%20State:%20Open). 
Please keep an open line of communication with us because we may have questions and comments on your changes. Otherwise, we won't be able to incorporate your contributions.

## Contribute to other libraries and create your own

Kotlin ships with the standard library providing core capabilities and a number of additional (kotlinx) libraries that extend its functionality. 
Each kotlinx library is developed in a separate repository, has its own versioning and release cycle. 

> If you want to contribute to the standard library, follow [the standard process for contributing](https://github.com/JetBrains/kotlin/blob/master/docs/contributing.md).

Reviewer 4: This seems to repeat the previous paragraph and also doesn't fall under the topic of the current paragraph. Maybe we don't need it here?

Author: Agree. Let's remove.
